File: ts/256/153.ts
export function findMin(nums: number[]): number {
  let l = 0, r = nums.length - 1
  if (nums[l] < nums[r]) return nums[l]
  while (l < r) {
    const mid = l + ((r - l) >> 1)
    if (nums[mid] < nums[r]) {
      r = mid
    } else {
      l = mid + 1
    }
  }
  return nums[l]
};

File: py3/256/153.py
from typing import List

class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        if nums[l] < nums[r]:
            return nums[l]
        while l < r:
            m = l + (r - l) // 2
            if nums[m] > nums[m + 1]:
                return nums[m + 1]
            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m
        return nums[l]